Word_Break.java

```diff
@@ -0,0 +1,66 @@
+package Week_3;
+
+import java.util.*;
+
+
+/*1.	PROBLEM STATEMENT:
+ * Given a non-empty string s and a dictionary wordDict containing a list of non-empty words, determine if s can be segmented into a space-separated sequence of one or more dictionary words. You may assume the dictionary does not contain duplicate words.
+  2.    EXAMPLE :
+  Consider the dictionary
+  { pear, salmon, foot, prints, footprints, leave, you, sun, girl, enjoy },
+	INPUTS								OUTPUT					EXPLANATION
+	youenjoy							true				can be segmented as "you enjoy"
+	youleavefootprints					true				can be segmented as "you leave footprints" or "you leave foot prints"
+	salmonenjoyapples					false				cannot be segmented 
+3. OPTIMIZATION
+	Using dynamic programming
+
+*/
+public class Word_Break {
+
+	public static void main(String[] args) {
+		Scanner s=new Scanner(System.in);
+		List<String> wordDict= new ArrayList<>();
+		int n=s.nextInt();
+		for(int i=0;i<n;i++) {
+			wordDict.add(s.next());
+		}
+		String str=s.next();
+		System.out.println(wordBreak(str,wordDict));
+	}
+	 public static boolean wordBreak(String s, List<String> wordDict) {

```

Comment:
Not a major issue, but from the problem statement, I would expect this function to take (String s, Set<String> wordDict).  In an in person it would be good to clarify this with your interviewer and make sure you are starting with the right function signature

Reply:
Yes I felt the same but this was the function signature required for leetcode submission